hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/ozone/container/common/TestEndPoint.java

```diff
@@ -165,17 +165,95 @@ public void testGetVersionTask() throws Exception {
     }
   }
 
+
+  /**
+   * Writing data to tmp dir and checking if it has been cleaned.
+   */
   @Test
-  public void testCheckVersionResponse() throws Exception {
+  public void testTmpDirCleanup() throws Exception {
     OzoneConfiguration conf = SCMTestUtils.getConf();
     conf.setBoolean(OzoneConfigKeys.DFS_CONTAINER_IPC_RANDOM_PORT,
         true);
     conf.setBoolean(OzoneConfigKeys.DFS_CONTAINER_RATIS_IPC_RANDOM_PORT,
         true);
-    conf.setBoolean(OzoneConfigKeys.DFS_CONTAINER_RATIS_DATASTREAM_ENABLED,
+    conf.setFromObject(new ReplicationConfig().setPort(0));
+    try (EndpointStateMachine rpcEndPoint = createEndpoint(conf,
+        serverAddress, 1000)) {
+      DatanodeDetails datanodeDetails = randomDatanodeDetails();
+      OzoneContainer ozoneContainer = new OzoneContainer(
+          datanodeDetails, conf, getContext(datanodeDetails), null);
+      rpcEndPoint.setState(EndpointStateMachine.EndPointStates.GETVERSION);
+
+      String clusterId = scmServerImpl.getClusterId();
+
+      // write some data before calling versionTask.call()
+      MutableVolumeSet volumeSet = ozoneContainer.getVolumeSet();
+      for (HddsVolume hddsVolume : StorageVolumeUtil.getHddsVolumesList(
+          volumeSet.getVolumesList())) {
+        hddsVolume.format(clusterId);
+        hddsVolume.createWorkingDir(clusterId, null);
+        hddsVolume.createDeleteServiceDir(clusterId);
+
+        // Create a container and move it under the tmp delete dir.
+        KeyValueContainer container =
+            setUpTestContainer(hddsVolume, clusterId, conf);
+        assertTrue(container.getContainerFile().exists());
+      }
+
+      VersionEndpointTask versionTask = new VersionEndpointTask(rpcEndPoint,
+          conf, ozoneContainer);
+      EndpointStateMachine.EndPointStates newState = versionTask.call();
+
+      Assertions.assertEquals(EndpointStateMachine.EndPointStates.REGISTER,
+          newState);
+
+      // assert that tmp dir is empty
+      for (HddsVolume hddsVolume : StorageVolumeUtil.getHddsVolumesList(
+          volumeSet.getVolumesList())) {
+        Assert.assertFalse(KeyValueContainerUtil.ContainerDeleteDirectory
+            .getDeleteLeftovers(hddsVolume).hasNext());
+      }
+    }
+  }
+
+  private KeyValueContainer setUpTestContainer(
```

Comment:
This looks  similar to `TestHDDSDatanodeService#setUpTestContainer`. Can we make these one method and move it to a common place like `ContainerTestUtils`?